src/lib/pickles/make_sponge.mli

```diff
@@ -1,8 +1,22 @@
+(** This module provides interfaces and functors to instantiate a hash function
+    built from a permutation and the {{
+    https://en.wikipedia.org/wiki/Sponge_function} sponge construction }.
+
+    The interfaces have been created to be used with permutations consisting of
+    partial and full rounds, like {{ https://eprint.iacr.org/2019/458.pdf}
+    Poseidon }.
```

Comment:
May also be good to mention any assumptions about whether the object being hashed are fixed-length or variable length with any padding applied.

Reply:
Opened https://github.com/MinaProtocol/mina/issues/14192.